tools/codegen/pkg/modelgen/model.go

```diff
@@ -217,6 +220,17 @@ func (m *Model) addTopLevelFields(fd *FileDescriptor) {
 	m.Comment = fmt.Sprintf("%s\n%s", fd.getComment(syntaxPath), fd.getComment(packagePath))
 }
 
+func validateTmplName(name string) error {
```

Comment:
I have been debating on how much to validate for template developers, especially because they would immediately see compile time error on the generated code if things don't work. It is not as good as getting explicit error on the template proto, but reasonable enough. For this case, I will add the checks you suggested. 

Rethinking about this, we are currently validating most of the stuff except the package name and template name. It would only make sense to validate those fields as well, and finish all the validation for template developers. I will make the change in the PR on Monday. Thanks for the push.